lib/tpm2_options.h

```diff
@@ -66,6 +66,7 @@ typedef bool (*tpm2_arg_handler)(int argc, char **argv);
  */
 #define TPM2_OPTIONS_NO_SAPI 0x1
 #define TPM2_OPTIONS_OPTIONAL_SAPI 0x2
+#define TPM2_OPTIONS_OPTIONAL_SAPI_AND_FAKE_TCTI 0x4
```

Comment:
Offhand, I would just make NO_SAPI the fake sapi, so if someone does something like
`tpm2_nvread --tcti=none --cphash xxx` it just works. But existing tools that check if ectx is null, like makecredential would need to have a flag set or something to not try to use the ectx.

How exactly do you see tpm2_nvread being executed? Do you envision that if --cphash is requested that you just send the fake tcti down and avoid users needing to do --tcti=none?

Reply:
We have to rely on tcti=none being called so as to not break compatibility with 4.X

Reply:
If tcti=none, I would just send the FAKE tcti and use it's magic to determine if it's fake tcti for things like makecred.